File: generate_icon.dart
import 'dart:io';
import 'dart:ui' as ui;
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

// This script generates app icons from Font Awesome icons
// Run: dart run generate_icon.dart

void main() async {
  // Initialize Flutter bindings
  WidgetsFlutterBinding.ensureInitialized();
  
  // Icon configuration
  const iconData = Icons.business_center; // Business briefcase icon
  const iconSize = 192.0;
  const iconColor = Color(0xFF2196F3); // Material Blue
  const backgroundColor = Colors.white;
  
  // Create main icon
  final mainIcon = await createIcon(
    iconData: iconData,
    size: 512,
    iconSize: iconSize * 2.0,
    iconColor: iconColor,
    backgroundColor: backgroundColor,
  );
  
  // Create foreground icon for adaptive icon
  final foregroundIcon = await createIcon(
    iconData: iconData,
    size: 512,
    iconSize: iconSize * 2.0,
    iconColor: iconColor,
    backgroundColor: Colors.transparent,
  );
  
  // Save icons
  await File('assets/icon/app_icon.png').writeAsBytes(mainIcon);
  await File('assets/icon/app_icon_foreground.png').writeAsBytes(foregroundIcon);
  
  print('Icons generated successfully!');
  print('- assets/icon/app_icon.png');
  print('- assets/icon/app_icon_foreground.png');
  print('\nNext steps:');
  print('1. Run: flutter pub get');
  print('2. Run: flutter pub run flutter_launcher_icons');
  
  exit(0);
}

Future<Uint8List> createIcon({
  required IconData iconData,
  required double size,
  required double iconSize,
  required Color iconColor,
  required Color backgroundColor,
}) async {
  final recorder = ui.PictureRecorder();
  final canvas = Canvas(
    recorder,
    Rect.fromLTWH(0, 0, size, size),
  );
  
  // Draw background
  if (backgroundColor != Colors.transparent) {
    final backgroundPaint = Paint()
      ..color = backgroundColor
      ..style = PaintingStyle.fill;
    canvas.drawRect(Rect.fromLTWH(0, 0, size, size), backgroundPaint);
  }
  
  // Draw icon
  final textPainter = TextPainter(
    text: TextSpan(
      text: String.fromCharCode(iconData.codePoint),
      style: TextStyle(
        fontSize: iconSize,
        fontFamily: iconData.fontFamily,
        color: iconColor,
      ),
    ),
    textDirection: TextDirection.ltr,
  );
  
  textPainter.layout();
  final offset = Offset(
    (size - textPainter.width) / 2,
    (size - textPainter.height) / 2,
  );
  textPainter.paint(canvas, offset);
  
  // Convert to image
  final picture = recorder.endRecording();
  final image = await picture.toImage(size.toInt(), size.toInt());
  final byteData = await image.toByteData(format: ui.ImageByteFormat.png);
  
  return byteData!.buffer.asUint8List();
}

File: generate_icon.py
#!/usr/bin/env python3
"""
Generate app icons for BizSync using Font Awesome icons
"""

from PIL import Image, ImageDraw, ImageFont
import os

# Create assets/icon directory if it doesn't exist
os.makedirs('assets/icon', exist_ok=True)

# Icon configuration
ICON_SIZE = 512
ICON_COLOR = (33, 150, 243)  # Material Blue
BG_COLOR = (255, 255, 255)  # White
FONT_SIZE = 320

def create_icon(size, icon_unicode, color, bg_color, transparent_bg=False):
    """Create an icon image"""
    # Create image with background
    if transparent_bg:
        image = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    else:
        image = Image.new('RGBA', (size, size), bg_color + (255,))
    
    draw = ImageDraw.Draw(image)
    
    # Draw a business/sync related shape (simplified briefcase)
    # Since we can't use actual Font Awesome, let's draw a simple business icon
    margin = size // 8
    icon_rect = [margin, margin * 2, size - margin, size - margin]
    
    # Draw briefcase shape
    # Main body
    draw.rounded_rectangle(
        [icon_rect[0], icon_rect[1] + size // 8, icon_rect[2], icon_rect[3]], 
        radius=size // 20, 
        fill=color + (255,),
        outline=None
    )
    
    # Handle
    handle_width = size // 3
    handle_x = (size - handle_width) // 2
    draw.arc(
        [handle_x, icon_rect[1] - size // 16, handle_x + handle_width, icon_rect[1] + size // 8],
        start=180, end=0,
        fill=color + (255,),
        width=size // 20
    )
    
    # Lock/latch in center
    lock_size = size // 8
    lock_x = (size - lock_size) // 2
    lock_y = icon_rect[1] + (icon_rect[3] - icon_rect[1]) // 3
    draw.rectangle(
        [lock_x, lock_y, lock_x + lock_size, lock_y + lock_size // 2],
        fill=bg_color + (255,)
    )
    
    return image

# Generate main icon
main_icon = create_icon(ICON_SIZE, '\\uf0b1', ICON_COLOR, BG_COLOR, transparent_bg=False)
main_icon.save('assets/icon/app_icon.png')

# Generate foreground icon for adaptive icon (transparent background)
foreground_icon = create_icon(ICON_SIZE, '\\uf0b1', ICON_COLOR, BG_COLOR, transparent_bg=True)
foreground_icon.save('assets/icon/app_icon_foreground.png')

print("Icons generated successfully!")
print("- assets/icon/app_icon.png")
print("- assets/icon/app_icon_foreground.png")
print("\nNext steps:")
print("1. Run: flutter pub get")
print("2. Run: flutter pub run flutter_launcher_icons")